docs/bin/plugin_formatter.py

```diff
@@ -247,6 +259,12 @@ def get_plugin_info(module_dir, limit_to=None, verbose=False):
         # use ansible core library to parse out doc metadata YAML and plaintext examples
         doc, examples, returndocs, metadata = plugin_docs.get_docstring(module_path, fragment_loader, verbose=verbose)
 
+        for key, opt in (doc.get('options') or {}).items():
```

Comment:
Why this particular formulation of doc['options']?  Do we have cases where doc['options'] does not exist?  Do we have cases where doc['options'] returns a non-dict?

Reply:
yes, as well as `None`

Reply:
That feels like the place we should be doing this, then.  We have a documentation schema we should be filling in an empty dict for options or failing to validate the documentation via the schema in this case.  (It looks like validate-modules does fail for None:
```
             if 'options' in doc and doc['options'] is None:
                    self.reporter.error(
                        path=self.object_path,
                        code=320,
                        msg='DOCUMENTATION.options must be a dictionary/hash when used',
                    )
```